pkg/ssh/ssh.go

```diff
@@ -163,11 +163,38 @@ func (d *realSSHDialer) Dial(network, addr string, config *ssh.ClientConfig) (*s
 	return ssh.Dial(network, addr, config)
 }
 
+// timeoutDialer wraps an sshDialer with a timeout around Dial().
```

Comment:
Can you add a comment either here or up above where it's used with a little more context on why this is needed?